app/src/pages/inside/settingsPage/settingTabs/demoDataTab/demoDataTabForm/demoDataTabForm.jsx

```diff
@@ -0,0 +1,133 @@
+import { activeProjectSelector } from 'controllers/user';
+import { showNotification, NOTIFICATION_TYPES } from 'controllers/notification';
+import { Input } from 'components/inputs/input';
+import { BigButton } from 'components/buttons/bigButton';
+import { FieldProvider } from 'components/fields/fieldProvider';
+import { FieldErrorHint } from 'components/fields/fieldErrorHint';
+import { SpinningPreloader } from 'components/preloaders/spinningPreloader';
+import styles from './demoDataTabForm.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  postfixInputPlaceholder: {
+    id: 'DemoDataTabForm.postfixInputPlaceholder',
+    defaultMessage: 'Enter postfix',
+  },
+  generateButtonTitle: {
+    id: 'DemoDataTabForm.generateButtonTitle',
+    defaultMessage: 'Generate demo data',
+  },
+  preloaderInfo: {
+    id: 'DemoDataTabForm.preloaderInfo',
+    defaultMessage:
+      'Data generation is started. The process can take several minutes, please wait.',
+  },
+  generateDemoDataSuccess: {
+    id: 'SuccessMessages.generateDemoDataSuccess',
+    defaultMessage: 'Demo data has been generated',
+  },
+});
+
+@reduxForm({
+  form: 'demoDataTabForm',
+  validate: ({ demoDataPostfix }) => ({
+    demoDataPostfix:
+      (!demoDataPostfix || !validate.demoDataPostfix(demoDataPostfix)) && 'demoDataPostfixHint',
+  }),
+})
+@connect(
+  (state) => ({
+    projectId: activeProjectSelector(state),
+  }),
+  {
+    showNotification,
+  },
+)
+@injectIntl
+export class DemoDataTabForm extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    handleSubmit: PropTypes.func.isRequired,
+    projectId: PropTypes.string.isRequired,
+    showNotification: PropTypes.func.isRequired,
+    reset: PropTypes.func.isRequired,
+  };
+
+  state = {
+    isLoading: false,
+  };
+
+  onFormSubmit = (formData) => {
+    const { intl, projectId, reset } = this.props;
+    const data = {
+      isCreateDashboard: 'true',
+      postfix: formData.demoDataPostfix,
+    };
+
+    this.setState({
+      isLoading: true,
+    });
+    fetch(URLS.generateDemoData(projectId), { method: 'POST', data })
+      .then(() => {
+        this.props.showNotification({
+          message: intl.formatMessage(messages.generateDemoDataSuccess),
+          type: NOTIFICATION_TYPES.SUCCESS,
+        });
+        this.setState({
+          isLoading: false,
+        });
+        reset('demoDataTabForm');
+      })
+      .catch((e) => {
+        const error = (e.response && e.response.data && e.response.data.message) || e.message;
+        this.props.showNotification({
+          messageId: 'failureDefault',
+          type: NOTIFICATION_TYPES.ERROR,
+          values: { error },
+        });
+        this.setState({
+          isLoading: false,
+        });
+      });
+  };
+
+  render() {
+    const { intl, handleSubmit } = this.props;
+
+    return (
+      <form onSubmit={handleSubmit(this.onFormSubmit)}>
```

Comment:
Please add a class `demo-data-form`

Reply:
We use it in this case.
See this comment:
https://github.com/reportportal/service-ui/pull/877#discussion_r210603367